x-pack/platform/plugins/shared/streams/server/lib/streams/helpers/migrate_on_read.ts

```diff
@@ -34,5 +40,32 @@ export function migrateOnRead(definition: Record<string, unknown>): Streams.all.
     };
     delete (migratedDefinition.ingest as { unwired?: {} }).unwired;
   }
+
+  // Migrate routing "if" Condition to Streamlang
+  if (
+    migratedDefinition.ingest &&
+    typeof migratedDefinition.ingest === 'object' &&
+    'wired' in migratedDefinition.ingest &&
+    (migratedDefinition.ingest as { wired?: unknown }).wired &&
+    typeof (migratedDefinition.ingest as { wired?: any }).wired === 'object' &&
+    Array.isArray((migratedDefinition.ingest as { wired?: any }).wired.routing) &&
+    (migratedDefinition.ingest as { wired?: any }).wired.routing.some((route: any) =>
+      isLegacyCondition(route.if)
+    )
+  ) {
+    migratedDefinition = migrateRoutingIfConditionToStreamlang(migratedDefinition);
+  }
+
+  // Migrate old flat processing array to Streamlang DSL
+  if (
+    migratedDefinition.ingest &&
+    typeof migratedDefinition.ingest === 'object' &&
+    Array.isArray((migratedDefinition.ingest as { processing?: unknown }).processing)
+  ) {
+    migratedDefinition = migrateOldProcessingArrayToStreamlang(migratedDefinition);
+  }
+
+  Streams.all.Definition.asserts(migratedDefinition as unknown as BaseStream.Definition);
```

Comment:
I removed this bit because it's quite expensive since it's happening _a lot_.

We need to make sure this migration on read is as cheap as possible. We probably shouldn't solve it on here, but ideally we manage to be able to remove it at some point as well...

For now please either do the assert only when it's actually migrating.

Reply:
Makes sense, I'll make sure this only runs if one of the migrators has run.